src/slurm_plugin/slurm_resources.py

```diff
@@ -136,61 +137,65 @@ def is_nodeaddr_set(self):
         """Check if slurm node is in any drain(draining, drained) states."""
-        return self.SLURM_SCONTROL_DRAIN_STATE in self.state
+        return self.SLURM_SCONTROL_DRAIN_STATE in self.states
 
     def is_drained(self):
         """
         Check if slurm node is in drained state.
 
         drained(sinfo) is equivalent to IDLE+DRAIN(scontrol) or DOWN+DRAIN(scontrol)
         """
-        return self._is_drain() and (
-            self.SLURM_SCONTROL_IDLE_STATE in self.state or self.SLURM_SCONTROL_DOWN_STATE in self.state
-        )
+        return self._is_drain() and (self.SLURM_SCONTROL_IDLE_STATE in self.states or self.is_down())
 
     def is_powering_down(self):
         """Check if slurm node is in powering down state."""
-        return self.SLURM_SCONTROL_POWERING_DOWN_STATE in self.state
+        return any(
+            powering_down_state in self.states for powering_down_state in self.SLURM_SCONTROL_POWERING_DOWN_STATES
+        )
 
     def is_power(self):
         """Check if slurm node is in power state."""
-        return self.SLURM_SCONTROL_POWER_STATE == self.state
+        return self.SLURM_SCONTROL_POWER_STATE == self.state_string
 
     def is_down(self):
         """Check if slurm node is in a down state."""
-        return self.SLURM_SCONTROL_DOWN_STATE in self.state and not self.is_powering_down()
+        return self.SLURM_SCONTROL_DOWN_STATE in self.states and not self.is_powering_down()
 
     def is_up(self):
         """Check if slurm node is in a healthy state."""
         return not self._is_drain() and not self.is_down() and not self.is_powering_down()
 
     def is_powering_up(self):
         """Check if slurm node is in powering up state."""
-        return self.SLURM_SCONTROL_POWER_UP_STATE in self.state
+        return self.SLURM_SCONTROL_POWER_UP_STATE in self.states
 
     def is_online(self):
         """Check if slurm node is online with backing instance."""
-        return self.state in self.SLURM_SCONTROL_ONLINE_STATES
+        return self.state_string in self.SLURM_SCONTROL_ONLINE_STATES
 
     def is_configuring_job(self):
         """Check if slurm node is configuring with job and haven't begun to run a job."""
         return self.is_powering_up() and self.has_job()
 
     def is_power_with_job(self):
         """Dynamic nodes allocated a job but power up process has not started yet."""
-        return self.state == self.SLURM_SCONTROL_POWER_WITH_JOB_STATE
+        return self.state_string == self.SLURM_SCONTROL_POWER_WITH_JOB_STATE
 
     def is_running_job(self):
         """Check if slurm node is running a job but not in configuring job state."""
         return not self.is_powering_up() and self.has_job() and not self.is_power_with_job()
 
     def is_resume_failed(self):
         """Check if node resume timeout expires."""
-        return self.state == self.SLURM_SCONTROL_RESUME_FAILED_STATE
+        return self.state_string == self.SLURM_SCONTROL_RESUME_FAILED_STATE
```

Comment:
one last comment, sorry for not calling out this earlier. Probably it's not even an actual problem at the moment but in case Slurm does not always guarantee the order in the state string we might end up with something like `DOWN+NOT_RESPONDING+CLOUD+POWERED_DOWN` which represents that same state but will fail our check.
To make this logic more robust we could represent the state string as a set and then check for sets equality.
```
SLURM_SCONTROL_RESUME_FAILED_STATE = {DOWN,CLOUD,POWERED_DOWN,NOT_RESPONDING}

return self.stats == self.SLURM_SCONTROL_RESUME_FAILED_STATE
```

Same applies to other few places

Reply:
Thank you! Changed to for SLURM_SCONTROL_RESUME_FAILED_STATE, SLURM_SCONTROL_POWER_WITH_JOB_STATE and SLURM_SCONTROL_POWER_STATE